src/gui/mzroll/gallerywidget.cpp

```diff
@@ -1,477 +1,449 @@
-	for(int i=0; i < groups.size(); i++ ) {
-		PeakGroup* group = groups[i];
-		if (group == NULL) continue;
-
-		float minmz = group->minMz;
-		float maxmz = group->maxMz;
-		Compound* c = group->getCompound();
-
-   		mzSlice slice(minmz,maxmz,rtmin,rtmax);
-		if(c) slice.compound = c;
-                if(c && !c->srmId().empty()) slice.srmId=c->srmId();
-
-		TinyPlot* plot = addEicPlot(slice);
-		if ( plot ) {
-                        plot->setTitle(QString(group->getName().c_str()));
-			plot->setData(0, QVariant::fromValue(slice));
-
-			for(int j=0; j < group->peakCount(); j++ ) {
-				plot->addPoint(group->peaks[j].rt, group->peaks[j].peakIntensity);
-			}
-		}
-	}
-	replot();
+pair<float, float> GalleryWidget::rtBounds()
+{
+    return make_pair(_minRt, _maxRt);
 }
 
+void GalleryWidget::setRtBounds(float minRt, float maxRt)
+{
+    _minRt = minRt;
+    _maxRt = maxRt;
+    _fillPlotData();
+    replot();
+}
 
-void GalleryWidget::addEicPlots(std::vector<mzLink>&links) {
+pair<float, float> GalleryWidget::intensityBounds()
+{
+    return make_pair(_minIntensity, _maxIntensity);
+}
 
-	clear();
+void GalleryWidget::setIntensityBounds(float minIntensity, float maxIntensity)
+{
+    _minIntensity = minIntensity;
+    _maxIntensity = maxIntensity;
+    _fillPlotData();
+    replot();
+}
 
-	std::vector<mzSample*>samples = mainwindow->getVisibleSamples();
-	if (samples.size() == 0 ) return;
+void GalleryWidget::addEicPlots(PeakGroup* group, MavenParameters* mp)
+{
+    clear();
+    if (group == nullptr || !group->hasSlice())
+        return;
 
-	MassCutoff *compoundMassCutoff=mainwindow->getUserMassCutoff();
-        mzSlice& current = mainwindow->getEicWidget()->getParameters()->getMzSlice();
+    // set min/max limits on slice to cover the entire EIC range
+    // this helps with baseline calculation as well as RT adjustment
+    mzSlice slice = group->getSlice();
+    slice.rtmin = numeric_limits<float>::max();
+    slice.rtmax = numeric_limits<float>::min();
+    for (mzSample* sample : group->samples) {
+        slice.rtmin = min(slice.rtmin, sample->minRt);
+        slice.rtmax = max(slice.rtmax, sample->maxRt);
+    }
+    _eics = PeakDetector::pullEICs(&slice, group->samples, mp);
+    sort(begin(_eics), end(_eics), [this](EIC* first, EIC* second) {
+        return first->sample->getSampleOrder()
+               < second->sample->getSampleOrder();
+    });
+
+    _maxIntensity = numeric_limits<float>::min();
+    _minRt = numeric_limits<float>::max();
+    _maxRt = numeric_limits<float>::min();
+    for (EIC* eic : _eics) {
+        if (eic == nullptr)
+            continue;
+
+        QColor color = QColor::fromRgbF(eic->sample->color[0],
+                                        eic->sample->color[1],
+                                        eic->sample->color[2],
+                                        0.5);
+
+        Peak* samplePeak = nullptr;
+        for (auto& peak : group->peaks) {
```

Comment:
`Peak* getPeak(mzSample* sample)` method exists for finding the peak by sample name.